src/lucky_record/associations.cr

```diff
@@ -3,16 +3,34 @@ module LuckyRecord::Associations
     LuckyRecord::Repo.settings.lazy_load_enabled
   end
 
-  macro has_many(type_declaration)
+  macro has_many(type_declaration, foreign_key = nil)
     {% assoc_name = type_declaration.var %}
 
     association table_name: :{{ assoc_name }}
```

Comment:
Does this also need to use a foreign_key option?

Reply:
Do you mean custom foreign keys on has_many assocations? I don't have a use for them but I imagine this could be useful for someone's use case.

Reply:
Never mind. I was thinking about this wrong. Looks great